### PR TITLE
Small tweaks to the Storage Access API data

### PR DESCRIPTION
**Pull request by chrisdavidmills**
#### Summary

https://github.com/mdn/browser-compat-data/pull/21336 updated the [Storage Access API](https://developer.mozilla.org/en-US/docs/Web/API/Storage_Access_API) compat data rather nicely. I created https://github.com/mdn/content/pull/30500 to update the SAA docs and introduce the concept of related website sets to MDN, and in doing so, I found a couple of details I thought worth including in the BCD.

So, here goes.

### CODE REVIEW

**Comment by chrisdavidmills** [comment]
FYI @tunetheweb

File: api/Document.json [hunk 1]
```diff
@@ -6136,7 +6136,8 @@
           "spec_url": "https://privacycg.github.io/requestStorageAccessFor/#dom-document-requeststorageaccessfor",
           "support": {
             "chrome": {
-              "version_added": "119"
+              "version_added": "119",
+              "notes": "Requires the calling top-level page and the embedded document storage access is being requested for to be part of the same <a href='https://developer.mozilla.org/docs/Web/API/Storage_Access_API/Related_website_sets'>related website set</a>."
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply api/Document.json]: LGTM. Thanks!

**Review by tunetheweb — APPROVED** [review]
LGTM!

File: api/Permissions.json [hunk 2]
```diff
@@ -706,6 +706,42 @@
...
       },
+      "permission_top-level-storage-access": {
+        "__compat": {
+          "description": "<code>top-level-storage-access</code> permission",
+          "spec_url": "https://privacycg.github.io/requestStorageAccessFor/#permissions-integration",
+          "support": {
+            "chrome": {
+              "version_added": "119"
+            },
+            "chrome_android": "mirror",
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply api/Permissions.json]: OK. I've updated it for now, but we'll try to get it checked.

@cfredric, do you know what Chrome and Chrome Android versions the `top-level-storage-access` permission feature is supported from?

[the remark this answers is not in the document]
Comment (cfredric) [reply api/Permissions.json]: The `top-level-storage-access` permission support in Chrome (tracked as [requestStorageAccessFor](https://chromestatus.com/feature/5122534152863744)) shipped alongside [RWS](https://chromestatus.com/feature/5640066519007232), which was enabled by default in Chrome 119 ([commit](https://chromiumdash.appspot.com/commit/c11f88a6cb1725a0066f7c79a41fb4b30a96a1f3)).

[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply api/Permissions.json]: Thanks! So the data as-is does sound about right.

**Review by Elchi3 — APPROVED** [review]
Thanks and sorry for the delay.

**Pull request merged**